libclamav/readdb.c

```diff
@@ -2943,6 +2943,30 @@ static int cli_loadcdb(FILE *fs, struct cl_engine *engine, unsigned int *signo,
     return CL_SUCCESS;
 }
 
+static cl_error_t setHash(const char *const token, uint8_t hashDest[SHA1_HASH_SIZE],
+                          const char *const varname, uint32_t line)
+{
+
+    cl_error_t ret               = CL_SUCCESS;
+    uint8_t hash[SHA1_HASH_SIZE] = {0};
+
+    if ((2 * SHA1_HASH_SIZE) != strlen(token)) {
+        cli_errmsg("cli_loadcrt: line %u: %s is not the appropriate length for a SHA1 Hash\n", (unsigned int)line, varname);
+        ret = CL_EMALFDB;
+        goto end;
+    }
+
+    if (cli_hex2str_to(token, (char *)hash, strlen(token) < 0)) {
+        cli_errmsg("cli_loadcrt: line %u: Cannot convert %s to binary string\n", (unsigned int)line, varname);
+        ret = CL_EMALFDB;
+        goto end;
+    }
+    memcpy(hashDest, hash, SHA1_HASH_SIZE);
+
+end:
```

Comment:
We should use the `done:` label instead of `end:`, so our new alloc-macros will work with it, and for convention's sake.